co2eq/flights/index.js

```diff
@@ -73,32 +86,53 @@ function distanceFromDuration(hour) {
   return averageSpeedFromDuration(hour) * hour;
 }
 
-function emissionsForShortOrLongHaul(distance, bookingClass, isShortHaul) {
-  return ((a(isShortHaul) * distance * distance) + (b(isShortHaul) * distance) + c(isShortHaul)) / (averageNumberOfSeats(isShortHaul) * passengerLoadFactor)
+function emissionsForShortOrLongHaul(distance, bookingClass, passengerLoadFactor, passengerToFreightRatio, isShortHaul) {
+
+  return ((a(isShortHaul) * distance * distance) + (b(isShortHaul) * distance) + c(isShortHaul))
+    / (averageNumberOfSeats(isShortHaul) * passengerLoadFactor)
     * passengerToFreightRatio(isShortHaul)
     * bookingClassWeightingFactor(bookingClass, isShortHaul)
     * ((fuelCo2Intensity * radiativeForcingMultiplier) + fuelPreProductionCo2Intensity)
     + (aircraftFactor * distance)
     + airportinfrastructureFactor;
 }
 
-function emissionsBetweenShortAndLongHaul(distance, bookingClass) {
+function emissionsBetweenShortAndLongHaul(distance, bookingClass, passengerLoadFactor, passengerToFreightRatio) {
   // Formula for inbetween short and long haul is a linear interpolation between
   // both hauls
-  const eMin = emissionsForShortOrLongHaul(shortHaulDistanceThreshold, bookingClass, true);
-  const eMax = emissionsForShortOrLongHaul(longHaulDistanceThreshold, bookingClass, false);
+  const eMin = emissionsForShortOrLongHaul(
+    shortHaulDistanceThreshold, bookingClass, true, passengerLoadFactor, passengerToFreightRatio
+  );
+  const eMax = emissionsForShortOrLongHaul(
+    longHaulDistanceThreshold, bookingClass, false, passengerLoadFactor, passengerToFreightRatio
+  );
   // x is between 0 (short haul) and 1 (long haul)
   const x = (distance - shortHaulDistanceThreshold) / (longHaulDistanceThreshold - shortHaulDistanceThreshold);
   return ((1 - x) * eMin) + (x * eMax);
 }
 
-function emissionsFromDistanceAndClass(distance, bookingClass) {
+function emissionsFromDistanceAndClass(distance, bookingClass, passengerLoadFactor, passengerToFreightRatio) {
   if (distance < shortHaulDistanceThreshold || distance > longHaulDistanceThreshold) {
     // Flight is eigher short or long (but not in between)
     const isShortHaul = distance < shortHaulDistanceThreshold;
-    return emissionsForShortOrLongHaul(distance, bookingClass, isShortHaul);
+    return emissionsForShortOrLongHaul(distance, bookingClass, passengerLoadFactor, passengerToFreightRatio, isShortHaul);
+  }
+  return emissionsBetweenShortAndLongHaul(distance, bookingClass, passengerLoadFactor, passengerToFreightRatio);
+}
+
+function getLoadFactors(activity) {
+  if (activity.departureAirportCode && activity.destinationAirportCode) {
+    const departureAirportRegion = airportIataCodeToRegion(activity.departureAirportCode);
+    const destinationAirportRegion = airportIataCodeToRegion(activity.destinationAirportCode);
+    if (departureAirportRegion !== '' && destinationAirportRegion !== '') {
```

Comment:
Why are we checking for emptyness here? Would it make sense to just check for nullity?

Reply:
Yes indeed :)